### PR TITLE
Merge fix

### PR DESCRIPTION
**Pull request by JeanMeijer**

### CODE REVIEW

File: Sources/App/frontend-webpack/src/SearchBar.vue [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    };
+  },
+  methods:{
+    search: function() {
+      this.$router.push('/');
+      let cmp = this;
+      cmp.currentlySearchingFor = cmp.searchText;
+      this.tweetList =   [];
+      cmp.startIndex = 0;
+      cmp.endIndex = 10;
```
Comment (bhichem) [inline Sources/App/frontend-webpack/src/SearchBar.vue]: what does 10 represents? is it a constant or a variable that could be changed

File: Sources/WebApi/hippo_web/api.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+        results.append({'keywords': new_terms.split(), 'tweets': new_results})
+        
+    
+    return jsonify(results)
+
+
+#generate (most general) synonym from terms/keyw
+#generate (most general) synonym from terms/keyw
+@app.route('/api/category/<terms>', methods=['GET'])
+def pick_category(terms):
```
Comment (bhichem) [inline Sources/WebApi/hippo_web/api.py:90]: Avoid writing very long functions

File: Sources/DataAnalysis/data_analysis/nlp.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+    return builder.build()
+
+
+#
+# idea_filter = create_string_match_engine(["", ""]) # want
+# product_filter = create_string_match_engine(["", ""]) # app startup
+
+
+def match_keywords_filter(text: str) -> bool:
```
Comment (bhichem) [inline Sources/DataAnalysis/data_analysis/nlp.py:63]: rewrite this or delete it if it is no longer needed

File: Sources/DataAnalysis/runprogram.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/python
+import sys
+
+from data_analysis.streamer import TwitterStreaming
+
+content = [
```
Comment (bhichem) [inline Sources/DataAnalysis/runprogram.py]: what does this array represent? add comments when things are unclear

File: Sources/App/Homepage/js/new-age.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  "use strict"; // Start of use strict
+
+  // Smooth scrolling using jQuery easing
+  $('a.js-scroll-trigger[href*="#"]:not([href="#"])').click(function() {
+    if (location.pathname.replace(/^\//, '') == this.pathname.replace(/^\//, '') && location.hostname == this.hostname) {
+      var target = $(this.hash);
+      target = target.length ? target : $('[name=' + this.hash.slice(1) + ']');
+      if (target.length) {
+        $('html, body').animate({
+          scrollTop: (target.offset().top - 48)
```
Comment (bhichem) [inline Sources/App/Homepage/js/new-age.js:11]: 48 magical number, avoid magical numbers and strings declare them somewhere

File: Sources/App/frontend-webpack/src/SearchBar.vue [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+  },
+  mounted() {
+    this.scroll(this.tweetList);
+  }
+}
+</script>
+
+<style scoped lang="css">
+  .search-box
+  {
```
Comment (bhichem) [inline Sources/App/frontend-webpack/src/SearchBar.vue]: be consist ant about whether or not curly braces are on new line or not

**Review by bhichem — CHANGES_REQUESTED** [review]
Please also edit the readme file and the requirements together with the architecture

File: Sources/DataAnalysis/data_analysis/nlp.py (hunk 3 above)
Comment (TieSoul) [reply Sources/DataAnalysis/data_analysis/nlp.py:63]: We have removed it in the next version.

**Comment by TieSoul** [comment]
We will make the requested changes in the next version of Hippo. The current `develop` branch is different enough from this version that we cannot include it in this pull request without drastically changing the content.

**Pull request merged**